source/minar.cpp

```diff
@@ -251,7 +251,7 @@ int minar::SchedulerData::start(){
 
             if(dispatch_tree.get_num_elements() > 0) {
                 CallbackNode *root = dispatch_tree.get_root();
-                now_plus_tolerance = wrapTime(now + root->tolerance);
+                now_plus_tolerance = wrapTime(now + root->tolerance/2);
```

Comment:
The current algorithm only consider one side of the tolerance: scheduled_time - tolerance. But if we consider both side, what can we do if (now > scheduled_time + tolerance)?